packages/calcite-components/src/components/carousel/carousel.tsx

```diff
@@ -0,0 +1,656 @@
+  //
+  //  Properties
+  //
+  // --------------------------------------------------------------------------
+
+  /**
+   * When `true`, the carousel will autoplay and controls will be displayed. When "paused" at time of initialization, the carousel will not autoplay, but controls will be displayed.
+   */
+  @Prop({ reflect: true }) autoplay: AutoplayType = false;
+
+  @Watch("autoplay")
+  autoplayWatcher(autoplay: boolean): void {
+    if (!autoplay) {
+      this.handlePause();
+    }
+  }
+
+  /**
+   * Specifies how and if the "previous" and "next" arrows are displayed.
+   */
+  @Prop({ reflect: true }) arrowType: ArrowType = "inline";
+
+  /**
+   * Specifies if the component's controls are positioned absolutely on top of slotted Carousel Items.
+   */
+  @Prop({ reflect: true }) controlOverlay = false;
+
+  /**
+   * When `true`, interaction is prevented and the component is displayed with lower opacity.
+   */
+  @Prop({ reflect: true }) disabled = false;
+
+  /**
+   * Accessible name for the component.
+   */
+  @Prop() label!: string;
+
+  /**
+   *  When `autoplay` is `true`, specifies in milliseconds the length of time to display each Carousel Item.
+   */
+  @Prop({ reflect: true }) autoplayDuration = DURATION;
+
+  /**
+   * Made into a prop for testing purposes only
+   *
+   * @internal
+   */
+  // eslint-disable-next-line @stencil-community/strict-mutable -- updated by t9n module
+  @Prop({ mutable: true }) messages: CarouselMessages;
+
+  /**
+   * Use this property to override individual strings used by the component.
+   */
+  // eslint-disable-next-line @stencil-community/strict-mutable -- updated by t9n module
+  @Prop({ mutable: true }) messageOverrides: Partial<CarouselMessages>;
+
+  @Watch("messageOverrides")
+  onMessagesChange(): void {
+    /* wired up by t9n util */
+  }
+
+  /**
+   * Made into a prop for testing purposes only
+   *
+   * @internal
+   */
+  // eslint-disable-next-line @stencil-community/strict-mutable -- updated by t9n module
+  @Prop({ mutable: true }) paused: boolean;
+
+  /**
+   * The component's selected `calcite-carousel-item`.
+   *
+   * @readonly
+   */
+  @Prop({ mutable: true }) selectedItem: HTMLCalciteCarouselItemElement;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  // --------------------------------------------------------------------------
+
+  connectedCallback(): void {
+    connectInteractive(this);
+    connectLocalized(this);
+    connectMessages(this);
+  }
+
+  componentDidLoad(): void {
+    setComponentLoaded(this);
+  }
+
+  componentDidRender(): void {
+    updateHostInteraction(this);
+  }
+
+  disconnectedCallback(): void {
+    disconnectInteractive(this);
+    disconnectLocalized(this);
+    disconnectMessages(this);
+    this.clearIntervals();
+  }
+
+  async componentWillLoad(): Promise<void> {
+    if ((this.autoplay === "" || this.autoplay) && this.autoplay !== "paused") {
```

Comment:
Can we add an explainer similar to [this](https://github.com/Esri/calcite-design-system/blob/main/packages/calcite-components/src/components/link/link.tsx#L122-L123)?